# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContextTypes.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 gfx/gl/GLContextTypes.h
--- a/gfx/gl/GLContextTypes.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/gfx/gl/GLContextTypes.h	Mon Jun 15 18:25:55 2020 +0000
@@ -18,17 +18,6 @@
 
 enum class OriginPos : uint8_t { TopLeft, BottomLeft };
 
-struct GLFormats {
-  GLenum color_texInternalFormat = 0;
-  GLenum color_texFormat = 0;
-  GLenum color_texType = 0;
-  GLenum color_rbFormat = 0;
-
-  GLenum depthStencil = 0;
-  GLenum depth = 0;
-  GLenum stencil = 0;
-};
-
 enum class CreateContextFlags : uint16_t {
   NONE = 0,
   REQUIRE_COMPAT_PROFILE = 1 << 0,
@@ -48,6 +37,14 @@
 };
 MOZ_MAKE_ENUM_CLASS_BITWISE_OPERATORS(CreateContextFlags)
 
+struct GLContextCreateDesc {
+  CreateContextFlags flags = CreateContextFlags::NONE;
+};
+
+struct GLContextDesc final : public GLContextCreateDesc {
+  bool isOffscreen = false;
+};
+
 } /* namespace gl */
 } /* namespace mozilla */
 